distribution/client.go

```diff
@@ -222,6 +222,18 @@ func (c *Client) GetModel(reference string) (types.Model, error) {
 	return model, nil
 }
 
+// GetRemoteModel returns a remote model by reference
+func (c *Client) GetRemoteModel(ctx context.Context, reference string) (types.ModelArtifact, error) {
```

Comment:
Why add `GetRemoteModel` here instead of directly using a `registry.Client`? Right now, all the `*distribution.Client` methods represent some interaction with the local store. Happy to talk it through since the current responsibilities aren't particularly well delineated at the moment. But I think it could be good to push in direction of having lower level APIs that work with artifacts (e.g read/write/delete) for destinations (e.g. registry, store) and then higher level APIs here that provide the backing for ModelManager (pull/push/prune).